forms-flow-api/src/formsflow_api/services/form_process_mapper.py

```diff
@@ -504,3 +504,31 @@ def export(  # pylint:disable=too-many-locals
             }
 
         raise BusinessException(BusinessErrorCode.INVALID_FORM_PROCESS_MAPPER_ID)
+
+    @staticmethod
+    def validate_form_name_path_title(request):
+        """Validate a form name by calling the external validation API."""
+        # Retrieve the parameters from the query string
+        title = request.args.get("title")
+        name = request.args.get("name")
+        path = request.args.get("path")
+
+        # Check if at least one query parameter is provided
+        if not (title or name or path):
+            return {"message": "At least one query parameter (title, name, path) must be provided."}
+
+        # Combine them into query parameters dictionary
+        query_params = f"title={title}&name={name}&path={path}&select=title,path,name"
+
+        # Initialize the FormioService and get the access token
+        formio_service = FormioService()
+        form_io_token = formio_service.get_formio_access_token()
+        # Call the external validation API
+        validation_response = formio_service.get_form_search(query_params, form_io_token)
+
+        # Check if the validation response has any results
+        if validation_response and len(validation_response) > 0:
```

Comment:
After receiving a response from Form.io, check if the data is not empty. If it's not, ensure that the _id of the first element matches the form ID passed in the payload (in some cases). This is important, especially when editing a form, because if the anonymous URL is being edited, this API needs to be called.

Reply:
@sumesh-aot @auslin-aot

Reply:
Agreed @shuhaib-aot

Reply:
@shuhaib-aot Please review this PR updates